### DISCUSSION TITLE
Adding Multi level commands

### DISCUSSION THREAD

**Discussion by vinaynkaranth** [python-cmd2/cmd2#1355] [Q&A] [answered]
How can I achieve multi level commands (nested commands) using cmd2?

The command/subcommand hierarchy may look something like this:

```
show -
         module1
                 counters  <args>
                 parameters
          module2
                configuration
                history
set 
      module1
            parameter <args>
            config  <args>

```

**Comment by tleonhardt** [comment, marked as answer]
See our [subcommands.py](https://github.com/python-cmd2/cmd2/blob/master/examples/subcommands.py) example for how to use subcommands. I think that will give you what you want.
